src/semgrep_agent/main.py

```diff
@@ -142,6 +146,14 @@ def main(
             f.to_dict(omit=constants.PRIVACY_SENSITIVE_FIELDS) for f in new_findings
         ]
         click.echo(json.dumps(output))
+    elif gitlab_output:
+        #output all new findings in Gitlab format
+        output = {
+            "$schema": "https://gitlab.com/gitlab-org/security-products/security-report-schemas/-/blob/master/dist/sast-report-format.json",
+            "version": "2.0",
+            "vulnerabilities": [f.to_gitlab() for f in new_findings],
+        }
+        click.echo(json.dumps(output))
```

Comment:
We could even write it directly to a file maybe, if that's what GitLab needs and skip using a pipe. Or do they understand results from stdout as well?

Reply:
File support is an option.  Gitlab artifact definitions look like the following:

```yaml
goose-sast:
  extends: .sast-analyzer
  image: kboberg/goose:latest
  tags:
    - security
  artifacts:
    reports:
      sast: gl-sast-report.json
  script:
    - /app/goose -target $CI_PROJECT_DIR -config /app/config/goose_rules.json -decisiontree -outputmode gitlab > gl-sast-report.json
````

Reply:
cool, let's leave it this way then if other tools tend to use `cmd > gl-sast-report.json` syntax as well 👍